### DISCUSSION TITLE
error when send registration confirmation email 'OrchardCore.Email.SmtpSettings O 1.5

### DISCUSSION THREAD

**Discussion by yassinsayed** [OrchardCMS/OrchardCore#13152] [Q&A]
when send confirmation email and support 

An unhandled exception occurred while processing the request.
AggregateException: One or more errors occurred. (No service for type 'OrchardCore.Email.SmtpSettings' has been registered.)
System.Threading.Tasks.Task.ThrowIfExceptional(bool includeTaskCanceledExceptions)

InvalidOperationException: No service for type 'OrchardCore.Email.SmtpSettings' has been registered.
Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)

**Comment by hishamco** [comment]
Does the `OC.Email` module enabled? If Yes, please the repoduce steps

**Comment by yassinsayed** [comment]
solved by commenting below line
var Smtpsettings = controller.HttpContext.RequestServices.GetRequiredService<SmtpSettings>();

i think this is deprecated

**Reply by hishamco** [reply]
You can injected directly on the controller
